src/main/java/com/ssafy/trip/member/service/MemberService.java

```diff
@@ -0,0 +1,30 @@
+package com.ssafy.trip.member.service;
+
+import com.ssafy.trip.member.domain.Member;
+import com.ssafy.trip.member.dto.MemberResponse;
+import com.ssafy.trip.member.repository.MemberRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class MemberService {
+    private final MemberRepository memberRepository;
+
+    @Transactional(readOnly = true)
+    public ResponseEntity<MemberResponse> findById(Long id) {
+        Member member = memberRepository.findById(id).orElse(null);
+        if(member == null) {
+            return ResponseEntity.notFound().build();
+        }
+        MemberResponse memberResponse = MemberResponse.builder()
+                .id(member.getId())
+                .nickname(member.getNickname())
+                .profileUrl(member.getProfileUrl())
+                .build();
+        return ResponseEntity.ok(memberResponse);
```

Comment:
현재 service에서 `ResponseEntity`를 반환하고 계시는데요. 이렇게 되면 비즈니스 로직을 처리하는 역할인 service가 역할이 너무 많아지지 않을까요?? 저는 [이 블로그](https://ksh-coding.tistory.com/102)를 보았는데, 타당하다고 생각해서 controller에서 response dto를 생성하고 `ResponseEntity`에 담아 반환을 하고 있습니다. 정호님 생각은 어떠신가요?

Reply:
`member == null`일 경우 controller에서 `null`을 반환 받아 if문으로 한번 더 처리해줘야 할 것 같은데 혹시 좋은 방법 있을까요?

Reply:
controller에 exception을 throw해도 될 것 같아요. 존재하지 않는 멤버 id를 요청으로 보낸 경우에 400 error를 보내면 좋을 것 같습니다. exception은 한번 찾아보시고 마땅한게 없다면 제가 한 것 처럼 커스텀해서 사용해도 될 것 같아요~
```suggestion
    public ResponseEntity<MemberResponse> findById(Long id) {
        Member member = memberRepository.findById(id).orElse(null);
        if(member == null) {
            throw exception;
        }
        MemberResponse memberResponse = MemberResponse.builder()
                .id(member.getId())
                .nickname(member.getNickname())
                .profileUrl(member.getProfileUrl())
                .build();
        return ResponseEntity.ok(memberResponse);
```

Reply:
areacharacter 패키지에 있는 수현님 코드 참고해서 Exception 추가해봤습니다~